### PR TITLE
Implement '==' operator for entries

### LINKED ISSUE #372: Implement ==
I think we should be able to compare entries with `==` just [like we do with hashes](https://ruby-doc.org/core/Hash.html#method-i-3D-3D).

```rb
Hash.new(dn: 'dc=example,dc=com') == Hash.new(dn: 'dc=example,dc=com')
# => true

Net::LDAP::Entry.new(dn: 'dc=example,dc=com') == Net::LDAP::Entry.new(dn: 'dc=example,dc=com')
# => false
```

### PR DESCRIPTION
**Pull request by erlliam**
Hey @HarlemSquirrel I was going through your profile and saw this repository. This issue seemed like something I could solve.

Resolves #372

### CODE REVIEW

File: lib/net/ldap/entry.rb [hunk 1]
```diff
@@ -196,4 +196,8 @@ def setter?(sym)
     sym.to_s[-1] == ?=
   end
   private :setter?
+
+  def ==(other)
+    return self['dn'] == other['dn']
```
Comment (HarlemSquirrel) [inline lib/net/ldap/entry.rb:201]: This is a great start but I think we also need to ensure that all the attributes are the same.

**Review by HarlemSquirrel — CHANGES_REQUESTED** [review]

File: lib/net/ldap/entry.rb [hunk 2]
```diff
@@ -196,4 +196,8 @@ def setter?(sym)
     sym.to_s[-1] == ?=
   end
   private :setter?
+
+  def ==(other)
+    return @myhash == other.to_h
```
Comment (esparta) [inline lib/net/ldap/entry.rb:201]:
```suggestion
    @myhash == other.to_h
```

Also, should this method check if the `other` is the same class?

**Comment by erlliam** [comment]
Should the equal operator work for sub classes of entries ? I was going to write `other.class == self.class` instead of `instance_of?`, but I felt like the latter is probably the better choice. Is there any difference between those two?

File: lib/net/ldap/entry.rb (hunk 2 above)
Comment (HarlemSquirrel) [reply lib/net/ldap/entry.rb]: While we don't really have an exposed way to create objects that inherit from Net::LDAP::Entry I would say those would be OK be OK to consider equal as far as Class goes. For [Ruby Hash objects](https://ruby-doc.org/core-2.7.2/Hash.html#method-i-3D-3D) `==` Returns true if all of the following are true:

> - object is a Hash object.
> - hash and object have the same keys (regardless of order).
> - For each key key, hash[key] == object[key].

**Comment by HarlemSquirrel** [comment]
While we don't really have an exposed way to create objects that inherit from Net::LDAP::Entry I would say those would be OK be OK to consider equal as far as Class goes. For [Ruby Hash objects](https://ruby-doc.org/core-2.7.2/Hash.html#method-i-3D-3D) `==` Returns true if all of the following are true:

> - object is a Hash object.
> - hash and object have the same keys (regardless of order).
> - For each key key, hash[key] == object[key].

File: lib/net/ldap/entry.rb [hunk 3]
```diff
@@ -196,4 +196,8 @@ def setter?(sym)
     sym.to_s[-1] == ?=
   end
   private :setter?
+
+  def ==(other)
+    return other.instance_of?(self.class) && @myhash == other.to_h
```
Comment (HarlemSquirrel) [inline lib/net/ldap/entry.rb:201]: We don't need an explicit return here.
```suggestion
    other.instance_of?(self.class) && @myhash == other.to_h
```

**Comment by HarlemSquirrel** [comment]
@erlliam `other.class == self.class` and `other.instance_of?(self.class)` are doing essentially the same thing. The late does feel more to the spirit of Ruby where we often prefer what sounds or looks more "natural".

**Review by HarlemSquirrel — APPROVED** [review]
Nice work!

**Pull request merged**